packages/server-core/src/user/avatar/avatar.hooks.ts

```diff
@@ -101,6 +103,23 @@ const ensureUserAccessibleAvatars = async (context: HookContext<AvatarService>)
   }
 }
 
+const checkUserHasPermissionOrIsOwner = async (context: HookContext<AvatarService>) => {
+  if (await checkScope(context.params.user!, 'globalAvatars', 'write')) {
```

Comment:
Ideally the response of await operation should be stored in a variable and then that variable should be used in if condition.